src/types/CodepointWidthDetector.cpp

```diff
@@ -9,392 +9,358 @@ namespace
     // used to store range data in CodepointWidthDetector's internal map
     struct UnicodeRange final
     {
-        unsigned int lowerBound;
-        unsigned int upperBound;
-        CodepointWidth width;
+        char32_t lowerBound;
+        char32_t upperBound : 31;
+        char32_t isAmbiguous : 1;

```

Comment:
This reduces the size of the table by 33%.

Reply:
I love this.